packages/studio/src/App.tsx

```diff
@@ -3,17 +3,29 @@ import ActionsBar from "./components/ActionsBar";
 import Toast from "./components/Toast";
 import PreviewWithUseComponents from "./components/PreviewWithUseComponents";
 import LeftSidebar from "./components/LeftSidebar";
+import { ViewportStyles } from "./components/Viewport/defaults";
+import { useRef, useState } from "react";
 
 export default function App() {
+  const previewRef = useRef<HTMLDivElement>(null);
```

Comment:
Instead of adding the viewport state here, at the top-level of the app, I think it could be beneficial to put this in the Studio Store. I'm thinking we create a `PagePreview` slice and add that to the Store. Right now, it would just contain viewport info. Later, we could migrate things like `setActiveEntityFile` and `setActivePage` over to this Slice as well. It'd be the source of truth for everything Preview related. 

I think this should ultimately simplify the code. We don't need to worry about passing these new props to `PreviewWithUseComponents` or `ActionsBar`. Instead, internally, Components could access/set the viewport with our `useStudioStore` hook.

Reply:
done I think. I put viewportDimensions + setViewportDimensions in the slice